internal/service/domain/tree/tree_test.go

```diff
@@ -0,0 +1,831 @@
+
+		expectedError := storage.ErrSensorNotFound
+
+		currentTree := getTestTrees()[0]
+		sensor := getTestSensors()[0]
+		currentTree.Sensor = sensor
+		treeCluster := getTestTreeClusters()[0]
+		currentTree.TreeCluster = treeCluster
+
+		// Mock expectations
+		treeRepo.EXPECT().GetByID(ctx, id).Return(currentTree, nil)
+		treeClusterRepo.EXPECT().GetByID(ctx, currentTree.TreeCluster.ID).Return(treeCluster, nil)
+		sensorRepo.EXPECT().GetByID(ctx, currentTree.Sensor.ID).Return(nil, expectedError)
+
+		// When
+		result, err := svc.Update(ctx, id, treeUpdate)
+
+		// Then
+		assert.Error(t, err)
+		assert.Nil(t, result)
+		assert.ErrorContains(t, err, expectedError.Error())
+	})
+	t.Run("should return error if updating tree fails", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		expectedError := errors.New("update failed")
+
+		currentTree := getTestTrees()[0]
+		treeCluster := getTestTreeClusters()[0]
+		currentTree.TreeCluster = treeCluster
+		sensor := getTestSensors()[0]
+		currentTree.Sensor = sensor
+
+		// Mock expectations
+		treeRepo.EXPECT().GetByID(ctx, id).Return(currentTree, nil)
+		treeClusterRepo.EXPECT().GetByID(ctx, currentTree.TreeCluster.ID).Return(treeCluster, nil)
+		sensorRepo.EXPECT().GetByID(ctx, currentTree.Sensor.ID).Return(sensor, nil)
+		treeRepo.EXPECT().Update(ctx, id,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).Return(nil, expectedError)
+
+		// When
+		result, err := svc.Update(ctx, id, treeUpdate)
+
+		// Then
+		assert.Error(t, err)
+		assert.Nil(t, result)
+		assert.EqualError(t, err, handleError(expectedError).Error())
+	})
+}
+
+func getTestTreeClusters() []*entities.TreeCluster {
+	now := time.Now()
+
+	return []*entities.TreeCluster{
+		{
+			ID:            1,
+			CreatedAt:     now,
+			UpdatedAt:     now,
+			Name:          "Cluster 1",
+			Address:       "123 Main St",
+			Description:   "Test description",
+			SoilCondition: entities.TreeSoilConditionLehmig,
+			Archived:      false,
+			Latitude:      float64Ptr(9.446741),
+			Longitude:     float64Ptr(54.801539),
+			Trees:         getTestTrees(),
+		},
+		{
+			ID:            2,
+			CreatedAt:     now,
+			UpdatedAt:     now,
+			Name:          "Cluster 2",
+			Address:       "456 Second St",
+			Description:   "Test description",
+			SoilCondition: entities.TreeSoilConditionSandig,
+			Archived:      false,
+			Latitude:      nil,
+			Longitude:     nil,
+			Trees:         []*entities.Tree{},
+		},
+	}
+}
+
+func getTestTrees() []*entities.Tree {
```

Comment:
You also need those test trees in the import-test case. What about creating a utils file with those test data?

Reply:
I created one. To use it in the import, we first need to merge the branches and then create a new issue for that.